include/pressio/ops/kokkos/ops_level2.hpp

```diff
@@ -63,65 +63,123 @@ namespace pressio{ namespace ops{
 //-------------------------------
 // specialize for op(A) = A
 //-------------------------------
-template < typename A_type, typename x_type, typename scalar_type, typename y_type>
+template <
+  class A_type, class x_type, class y_type,
+  class alpha_t, class beta_t
+  >
 ::pressio::mpl::enable_if_t<
+  // level2 common constraints
+  ::pressio::Traits<A_type>::rank == 2 and
+  ::pressio::Traits<x_type>::rank == 1 and
+  ::pressio::Traits<y_type>::rank == 1 and
+  // TPL/container specific
   (::pressio::is_native_container_kokkos<A_type>::value or
    ::pressio::is_expression_acting_on_kokkos<A_type>::value) and
   (::pressio::is_native_container_kokkos<x_type>::value or
    ::pressio::is_expression_acting_on_kokkos<x_type>::value) and
   (::pressio::is_native_container_kokkos<y_type>::value or
     ::pressio::is_expression_acting_on_kokkos<y_type>::value) and
-  ::pressio::Traits<A_type>::rank == 2 and
-  ::pressio::Traits<x_type>::rank == 1 and
-  ::pressio::Traits<y_type>::rank == 1
+  // scalar compatibility
+  ::pressio::all_have_traits_and_same_scalar<A_type, x_type, y_type>::value and
+  std::is_convertible<alpha_t, typename A_type::non_const_value_type>::value and
```

Comment:
see comment below, i am surprised this overload gets selected since it should compile

Reply:
also why you are not using the Traits::scalar_type like you did for Eigen?

Reply:
good point, thanks for catching this: fixed with 1fb60febaea4bbed9207bc3e793fc98fc8d24e6b